code/modules/reagents/reagent_containers/chem_heirloom.dm

```diff
@@ -24,25 +25,21 @@
 	var/datum/component/heirloom/H = GetComponent(/datum/component/heirloom)
 	desc = H ? "[ishuman(H.owner) ? "The [H.family_name]" : "[H.owner.name]'s"] family's long-cherished wish is to open this bottle and get its chemical outside. Can you make that wish come true?" : "A hard locked bottle of [initial(rand_cont.name)]."
 
-/obj/item/reagent_containers/glass/chem_heirloom/afterattack(obj/target, mob/user, proximity)
-	if(!locked)
-		..()
-	return
-
-/obj/item/reagent_containers/glass/chem_heirloom/attackby(obj/item/I, mob/user, params)
-	if(!locked)
-		..()
-	return
-
 /obj/item/reagent_containers/glass/chem_heirloom/proc/unlock()
 	if(!locked) //A little bird said this would be an issue if goober-min tried to call this twice.
 		return
 	if(isliving(loc))
 		var/mob/living/M = loc
 		to_chat(M, "<span class='notice'>The [src] unlocks!</span>")
-	reagents.add_reagent(rand_cont, volume) //Add reagents
+	volume = 100
 	item_state = "hard_locked_open"
 	icon_state = "hard_locked_open"
 	locked = FALSE
 	spillable = TRUE
 	reagent_flags = OPENCONTAINER
+	say("Add [rand_cont] [volume]")
+	reagents.add_reagent(rand_cont, volume) //Add reagents
+
+/obj/item/reagent_containers/glass/chem_heirloom/Destroy()
+	. = ..()
+	qdel(roundend_callback)
```

Comment:
You would have to remove it from the roundend callback list you added it to when you called OnRoundend unless the subsystem registers for qdel already.

Reply:
I suppose it would but, better safe than sorry.